src/hooks/useComparison.ts

```diff
@@ -0,0 +1,47 @@
+import { useDispatch } from 'react-redux';
+
+import { RootState } from '../common/store';
+import { comparisonResults as secondRevisionResults } from '../mockData/9d5066525489';
+import { comparisonResults as thirdRevisionResults } from '../mockData/a998c42399a8';
+import { comparisonResults as firstRevisionResults } from '../mockData/bb6a5e451dac';
+import { setCompareData } from '../reducers/CompareResults';
+import { updateComparison } from '../reducers/ComparisonSlice';
+import { Strings } from '../resources/Strings';
+import { truncateHash } from '../utils/helpers';
+import { useAppSelector } from './app';
+
+const useComparison = () => {
+  const dispatch = useDispatch();
+  const comparisonResults = firstRevisionResults.concat(
+    secondRevisionResults,
+    thirdRevisionResults,
+  );
+  const activeComparison: string = useAppSelector(
+    (state: RootState) => state.comparison.activeComparison,
+  );
+  const allRevisions =
+    Strings.components.comparisonRevisionDropdown.allRevisions;
+
+  const switchComparisonData = () => {
+    if (activeComparison === truncateHash(firstRevisionResults[0].new_rev)) {
+      dispatch(setCompareData({ data: firstRevisionResults }));
+    }
+    if (activeComparison === truncateHash(secondRevisionResults[0].new_rev)) {
+      dispatch(setCompareData({ data: secondRevisionResults }));
+    }
+    if (activeComparison === truncateHash(thirdRevisionResults[0].new_rev)) {
+      dispatch(setCompareData({ data: thirdRevisionResults }));
+    }
+    if (activeComparison === allRevisions) {
```

Comment:
I think the "All revisions" option should use a key that wouldn't be translated (the Strings will be localized in the future). This key will never change and can be compared in the event handler. Then the (hardcoded) key is what is compared, not the string.

Reply:
[PCF-287 Change string to key for 'All revisions' option](https://mozilla-hub.atlassian.net/browse/PCF-287)